kin-sdk/kin-sdk-lib/src/test/java/kin/sdk/TransactionTest.java

```diff
@@ -0,0 +1,100 @@
+        Account account = new Account(source, sequenceNumber);
+        kin.base.Transaction baseTransaction = new kin.base.Transaction.Builder(account)
+            .addOperation(new CreateAccountOperation.Builder(destination, "2000").build())
+            .addFee(100)
+            .build();
+        baseTransaction.sign(source);
+        TransactionId id = new TransactionIdImpl(Utils.byteArrayToHex(baseTransaction.hash()));
+        transaction = new Transaction(id, baseTransaction);
+    }
+
+    private void mockServer() throws IOException {
+        mockWebServer = new MockWebServer();
+        mockWebServer.start();
+    }
+
+    @Test
+    public void getTransactionEnvelope_success() throws Exception {
+        mockWebServer.enqueue(TestUtils.generateSuccessMockResponse(this.getClass(), "tx_account_from.json"));
+        mockWebServer.enqueue(TestUtils.generateSuccessMockResponse(this.getClass(), "tx_account_to.json"));
+
+        String transactionEnvelope = transaction.getTransactionEnvelope();
+        kin.base.Transaction transaction2 = kin.base.Transaction.fromEnvelopeXdr(transactionEnvelope);
+
+        assertThat(transaction.getBaseTransaction().getSourceAccount().getAccountId(),
+            equalTo(transaction2.getSourceAccount().getAccountId()));
+        assertThat(transaction.getBaseTransaction().getSequenceNumber(), equalTo(transaction2.getSequenceNumber()));
+        assertThat(transaction.getBaseTransaction().getFee(), equalTo(transaction2.getFee()));
+        assertThat(transaction.getTransactionEnvelope(), equalTo(transaction2.toEnvelopeXdrBase64()));
+    }
+
+    @Test
+    public void decodeTransaction_success() throws Exception {
+        mockWebServer.enqueue(TestUtils.generateSuccessMockResponse(this.getClass(), "tx_account_from.json"));
+        mockWebServer.enqueue(TestUtils.generateSuccessMockResponse(this.getClass(), "tx_account_to.json"));
+
+        String transactionEnvelope = transaction.getTransactionEnvelope();
+        Transaction transaction2 = Transaction.decodeTransaction(transactionEnvelope);
+        assertThat(transaction.getSource().getAccountId(), equalTo(transaction2.getSource().getAccountId()));
+        assertThat(transaction.getBaseTransaction().getSequenceNumber(),
+            equalTo(transaction2.getBaseTransaction().getSequenceNumber()));
+        assertThat(transaction.getFee(), equalTo(transaction2.getFee()));
+        assertThat(transaction.getTransactionEnvelope(), equalTo(transaction2.getTransactionEnvelope()));
+    }
+
+    @Test
+    public void addSignature_success() {
+        KinAccountImpl mockKinAccount = mock(KinAccountImpl.class);
+        when(mockKinAccount.getKeyPair()).thenReturn(KeyPair.fromSecretSeed(SECRET_SEED_FROM));
+
+        assertThat(transaction.getBaseTransaction().getSignatures().size(), equalTo(1));
+        transaction.addSignature(mockKinAccount);
+        assertThat(transaction.getBaseTransaction().getSignatures().size(), equalTo(2));
```

Comment:
compare content also (with hardcoded expected values)

Reply:
But I need that one side will not be hardcoded, here I already have one side of the asserts that is hardcoded.
Explain again.